src/utils/hooks.js

```diff
@@ -19,3 +24,92 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+export function useUrgency(items) {
+	const [urgencyObject, setUrgecyObject] = useState({
+		overdue: [],
+		soon: [],
+		kindOfSoon: [],
+		notSoon: [],
+		inactive: [],
+	});
+
+	const getUrgency = (name) => {
+		const statusArray = Object.entries(urgencyObject).find(
+			([status, items]) => {
+				if (items.some((item) => item.name === name)) {
+					return status;
+				}
+			},
+		);
+		if (!statusArray) {
+			throw new Error(`Failed to get class name of ${name}`);
+		}
+		return statusArray[0];
+	};
+
+	const sortByUrgency = (item, daysUntilNextPurchase) => {
+		if (daysUntilNextPurchase < 0) {
+			return 'overdue';
+		} else if (daysUntilNextPurchase < 7) {
+			return 'soon';
+		} else if (daysUntilNextPurchase >= 7 && daysUntilNextPurchase < 30) {
+			return 'kindOfSoon';
+		} else if (daysUntilNextPurchase >= 30) {
+			return 'notSoon';
+		} else {
+			throw new Error(`Failed to place [${item.name}]`);
+		}
+	};
+
+	const calculateUrgency = (item) => {
+		// get date the item was purchased or created
+		const itemDate = getDateLastPurchasedOrDateCreated(
+			item.dateLastPurchased,
+			item.dateCreated,
+		);
+		// check how many days have passed since that date
+		const daysToToday = getDaysFromDate(itemDate);
+
+		// if more than 60 days have passed
+		if (daysToToday >= 60) {
+			// sort as inactive
+			return 'inactive';
+		} else {
+			// sort by the amount of days until next purchase date
+			const daysUntilNextPurchase = getDaysBetweenDates(
+				new Date(),
+				item.dateNextPurchased.toDate(),
+			);
+			return sortByUrgency(item, daysUntilNextPurchase);
+		}
+	};
+
+	useEffect(() => {
+		if (items.length === 0) return;
+
+		let initialUrgencyState = {
+			overdue: new Set(),
+			soon: new Set(),
+			kindOfSoon: new Set(),
+			notSoon: new Set(),
+			inactive: new Set(),
+		};
```

Comment:
It seems like the `urgencyState` is being initialized twice: first as an object with arrays during declaration and then re-initialized with Sets inside the useEffect. Could we use Sets from the start?